src/month.rs

```diff
@@ -119,3 +117,37 @@ impl fmt::Display for Month {
         })
     }
 }
+
+impl From<Month> for u8 {
+    fn from(month: Month) -> Self {
+        match month {
+            January => 1,
+            February => 2,
+            March => 3,
+            April => 4,
+            May => 5,
+            June => 6,
+            July => 7,
+            August => 8,
+            September => 9,
+            October => 10,
+            November => 11,
+            December => 12,
+        }
+    }
+}
+
+impl TryFrom<u8> for Month {
+    type Error = ComponentRange;
+
+    fn try_from(value: u8) -> Result<Self, ComponentRange> {
+        let n = NonZeroU8::new(value).ok_or_else(|| ComponentRange {
+            name: "month",
+            minimum: 1,
+            maximum: 12,
+            value: value.into(),
+            conditional_range: false,
+        })?;
+        Self::from_number(n)
```

Comment:
```suggestion
        NonZeroU8::new(value).map_or_else(Self::from_number, || Err(ComponentRange {
            name: "month",
            minimum: 1,
            maximum: 12,
            value: value as _,
            conditional_range: false,
        }))
```

I haven't checked this, but I'm pretty sure it'll compile.

Reply:
This one fails to compile because the first closure in `map_or_else` takes no arguments, but `from_number` takes an `NonZeroU8`.

Reply:
Ah, right. Flip the order of the two arguments. I forgot the "else" bit comes first.